pilot/pkg/model/service.go

```diff
@@ -422,6 +422,18 @@ func (h Hostname) String() string {
 	return string(h)
 }
 
+// Namespace returns the namespace extracted from hostname.
+// - k8s: the second part of the "." separated hostname. e.g. "default" for hostname
+//        "productpage.default.svc.cluster.local"
+// - other or if invalid format: an empty string
+func (h Hostname) Namespace() string {
+	split := strings.Split(h.String(), ".")
+	if len(split) >= 2 {
+		return split[1]
+	}
+	return ""
+}
+
```

Comment:
This restricts it only to kubernetes. You can’t even use it for VMs that have custom dns names.

Reply:
Is that the intention here ? That RBAC is only for kubernetes services ?

Reply:
I believe `namespace` is carried through because RBAC only allows short names and relies on k8s RBAC to enforce ACLs. Would we still need to propogate namespace through the model if host were always fully qualified?

Reply:
For the Istio RBAC, we currently only supports k8s as it relies on the concept of namespace and service. cc @liminw.

I first inline the code inside the plugin, but per @diemtvu's comment I moved it to be inside Hostname class thinking it might be useful for other code.

Reply:
Here is one option:
Add a function called GetServiceNamespace() to the service model. Implement it for each platform. For kube, the serviceNamespace returns the namespace from the DNS name. For other platforms, it returns a dummy "default" value, leaving the door open for other folks like CloudFoundry to add a different solution by which they can assign a namespace to the service.